python/lsst/afw/display/saturated.cc

```diff
@@ -0,0 +1,176 @@
+/**
+ * \file
+ *
+ * Handle saturated pixels when making colour images
+ */
+#include "boost/format.hpp"
+#include "lsst/utils/ieee.h"
+#include "lsst/afw/detection.h"
+#include "lsst/afw/image/MaskedImage.h"
+#include "Rgb.h"
+
+namespace lsst { namespace afw { namespace display {
+
+namespace {
+    template <typename ImageT>
+    class SetPixels : public detection::FootprintFunctor<ImageT> {
+    public:
+        SetPixels(ImageT const& img     // The image the source lives in
+                 ) : detection::FootprintFunctor<ImageT>(img), _value(0) {}
+
+        void setValue(float value) { _value = value; }
+
+        // method called for each pixel by apply()
+        void operator()(typename ImageT::xy_locator loc,        // locator pointing at the pixel
+                        int,                                    // column-position of pixel
+                        int                                     // row-position of pixel
+                       ) {
+            *loc = _value;
+        }
+    private:
+        float _value;
```

Comment:
You've got `float` here, and below you only instantiate for floating-point images, so why not drop the pretense (`template`) of supporting other types?